### ISSUE TITLE
undefined symbol: SSL_get1_peer_certificate

### ISSUE THREAD

**Issue by 0xReagan8** [memgraph/gqlalchemy#294]
**Memgraph version** Which version did you use?
2.13.0

**Environment** Some information about the environment you are using Memgraph
on: operating system, how do you connect, with or without docker, which driver
etc.
docker w/ conda pytohn =3.10.0

**Describe the bug** A clear and concise description of what the bug is.
trying to install 

undefined symbol: SSL_get1_peer_certificate

**To Reproduce** Steps to reproduce the behavior:
from gqlalchemy import Memgraph

1. Run the following query '...'

**Expected behavior** A clear and concise description of what you expected to
happen.

**Logs** If applicable, add logs of Memgraph, CLI output or screenshots to help
explain your problem.

**Additional context** Add any other context about the problem here.

**Comment by katarinasupe** [comment]
Hi @0xReagan8, can you try installing all of the [prerequisites](https://github.com/memgraph/gqlalchemy?tab=readme-ov-file#prerequisites), including the [build prerequisites](https://memgraph.github.io/pymgclient/introduction.html#build-prerequisites) for pymgclient? OpenSSL might be missing.

**Comment by 0xReagan8** [comment]
The following packages will be UPDATED:

ca-certificates    anaconda::ca-certificates-2023.08.22-~ --> conda-forge::ca-certificates-2023.11.17-hbcca054_0 
openssl               anaconda::openssl-3.0.12-h7f8727e_0 --> conda-forge::openssl-3.2.0-hd590300_1 

python-devtools           0.12.2             pyhd8ed1ab_0    conda-forge
cmake                     3.22.1               h1fce559_0    anaconda
make                      4.2.1                h1bed415_1    anaconda
gcc                       13.2.0               h574f8da_2    conda-forge
  gcc_impl_linux-64         13.2.0               h338b0a0_3    conda-forge
  libgcc-devel_linux-64     13.2.0             ha9c7c90_103    conda-forge
  libgcc-ng                 13.2.0               h807b86a_3    conda-forge
gxx                       13.2.0               h574f8da_2    conda-forge
  gxx_impl_linux-64         13.2.0               h338b0a0_3    conda-forge
openssl                   3.2.0                hd590300_1    conda-forge


I am still getting -> undefined symbol: SSL_get1_peer_certificate

**Comment by katarinasupe** [comment]
Can you write down the OS you're running this on? 
I see you're using Docker w/ Conda and Python 3.10.
Please also list the commands you run and at which step you get stuck. This will help me debug more quickly. I think this has something to do with the fact that Conda comes with its version of SSL.

**Comment by katarinasupe** [comment]
I see we had this issue a couple of times reported now, and it is mostly related to Conda env. Can you try installing gqlalchemy without Conda? Here is related discussion: https://github.com/memgraph/gqlalchemy/issues/285#issuecomment-1735737229

**Comment by as51340** [comment]
Investigated the issue, you are probably having locally old version of `pymgclient`. Clean `pymgclient` from global path (pip uninstall pymgclient gqlalchemy), activate your conda environmen and install ONLY gqlalchemy (pip install gqlalchemy). You should see that pymgclient 1.3.1 is being built while installing gqlalchemy. No need to install pymgclient before gqlalchemy, we will correct docs.

**Comment by katarinasupe** [comment]
@0xReagan8 can you try what @as51340 suggested and let us know whether that works for you 🙏 😄

**Comment by Cazual95** [comment]
> @0xReagan8 can you try what @as51340 suggested and let us know whether that works for you 🙏 😄

I tried this and it did not work for me. I am still encountering this issue.

**Comment by Cazual95** [comment]
I got it working.

I shouldve checked out this thread sooner: https://github.com/memgraph/pymgclient/issues/40

Installing openssl via conda forge and uninstalling gqlalchemy and pymgclient then reinstalling did it for me.

**Comment by katarinasupe** [comment]
I am glad to hear, @Cazual95, that it's working for you now, and thank you for providing the info on how you made it work to help others as well. What are you working on?

**Comment by Cazual95** [comment]
Of course! Im building a little app to experiment with streaming data. I want to test some real-time entity alignment algorithms. And maybe play with an event-driven graph.

**Comment by katarinasupe** [comment]
I somehow totally missed this comment. Memgraph seems like a good choice for such a graph. If you still haven't, join our [Discord](https://discord.gg/memgraph) community or schedule an [office hours](https://memgraph.com/office-hours) call if you have any questions.
